python/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![deny(warnings)]
+#![allow(clippy::borrow_deref_ref)]
```

Comment:
curious which line is violating this error? we should probably fix that instead.

Reply:
The issue is directly related to pyo3.  There's already a [ticket](https://github.com/PyO3/pyo3/issues/2555) filed, and it appears to be fixed in main. Once the fixed is release this clippy exception shouldn't be necessary.

Reply:
As a follow-up before merging, could you please add an issue to delete this clippy exception later?

Reply:
Second to what @fvaleye said, would be good to add a follow up issue on github. Also a good practice to add comment in the code for temp workarounds that we plan to remove in the future.